command.go

```diff
@@ -25,3 +25,20 @@ func (t OnOffCommand) Execute(ctx Context, args map[string]interface{}) proto.De
 func (t OnOffCommand) Name() string {
 	return "action.devices.commands.OnOff"
 }
+
+// OpenCloseCommand defines how a function should handle this specific trait
+type OpenCloseCommand func(ctx Context, params interface{}) proto.DeviceError
```

Comment:
this should be ```type OpenCloseCommand func(ctx Context, openPercent float, openDirection OpenCloseTraitDirection) proto.DeviceError```

there should also be a 
``` const OpenCloseTraitDirectionNone OpenCloseTraitDirection = "" ```

Reply:
done

Reply:
How about something like this

```
// OpenCloseCommand defines how a function should handle this specific trait
type OpenCloseCommand func(ctx Context, openPercent float64, openDirection OpenCloseTraitDirection) proto.DeviceError

// Execute validates the request and calls the user defined handler for the device trait
func (t OpenCloseCommand) Execute(ctx Context, args map[string]interface{}) proto.DeviceError {
	// validate if our EXECUTE Request contains the "openPercent" object so the handler can actually handle something
	openDirection := OpenCloseTraitDirectionNone
	if argOpenPercent, ok := args["openPercent"]; ok {
		if openPercent, ok := argOpenPercent.(float64); ok {
			// if openDirection is specified, set it, otherwise lets use none
			if argDir, ok := args["openDirection"]; ok {
				if dir, ok := argDir.(string); ok {
					openDirection = OpenCloseTraitDirection(dir)
				}
			}
			return t(ctx, openPercent, openDirection)
		}
		return proto.ErrorCodeNotSupported
	}
	return proto.ErrorCodeProtocolError
}
```

Reply:
Okay, this is a good idea.